Foreman/DataCache.cs

```diff
@@ -1084,58 +1093,108 @@ private static Item FindOrCreateUnknownItem(String itemName)
-                                //    ErrorLogging.LogLine(Convert.ToString(k) + ": " + Convert.ToString(tints[k]));
-                                //}
-                                //ErrorLogging.LogLine(Convert.ToString(((LuaTable)s)["icon"]));
-                                fileName = Convert.ToString(((LuaTable)s)["icon"]);
+                                LuaTable iconTable = (LuaTable)icons[iconKey];
+                                int actualIconSize = Convert.ToInt32(Convert.ToDouble(iconTable["icon_size"]) * Convert.ToDouble(iconTable["scale"]));
+                                // create and scale image
+                                Bitmap iconImage = new Bitmap
+                                (
+                                    LoadImage(Convert.ToString(iconTable["icon"])),
+                                    actualIconSize,
+                                    actualIconSize
+                                );
+                                // tint image
+                                LuaTable tintTable = (LuaTable)iconTable["tint"];
+                                if (tintTable != null)
+                                {
+                                    Color tintColor = Color.FromArgb
+                                    (
+                                        Convert.ToInt32(Convert.ToDouble(tintTable["a"]) * 255),
+                                        Convert.ToInt32(Convert.ToDouble(tintTable["r"]) * 255),
+                                        Convert.ToInt32(Convert.ToDouble(tintTable["g"]) * 255),
+                                        Convert.ToInt32(Convert.ToDouble(tintTable["b"]) * 255)
+
+                                    );
+                                    for (int i = 0; i < actualIconSize; i++)
+                                    {
+                                        for (int j = 0; j < actualIconSize; j++)
+                                        {
+                                            if (iconImage.GetPixel(i, j).A != 0)
+                                            {
+                                                iconImage.SetPixel(i, j, tintColor);
+                                            }
+                                        }
+                                    }
+                                }
+                                // offset image
+                                int offsetX = 0;
+                                int offsetY = 0;
+                                try
+                                {
+                                    LuaTable offsetTable = (LuaTable)iconTable["shift"];
+                                    offsetX = Convert.ToInt32(offsetTable["1"]);
+                                    offsetY = Convert.ToInt32(offsetTable["2"]);
+                                }
+                                catch (Exception e)
+                                {
+                                    // whatever

```

Comment:
Please include error logging here.

Reply:
sometimes there are no tables for "shift" or "tint" or whatever and since NLua kinda sucks, i can only check for it with try/catch and then check if it's == null
so i really wouldn't go and catch those errors, bloats my errorlog to ~200KB